### PR TITLE
add llama example

### PR DESCRIPTION
**Pull request by prathikr**
This PR adds run_classification.py example with ONNX Runtime accelerator classes. The example was taken from [Hugging Face's example](https://github.com/huggingface/transformers/blob/main/examples/pytorch/text-classification/run_classification.py) and was modified to use ORTTrainer/ORTTrainingArguments. Additionally, the PR adds observed performance metrics for `meta-llama/Llama-2-7b-hf`.

### CODE REVIEW

**Comment by prathikr** [comment]
@JingyaHuang can you please review?

File: examples/onnxruntime/training/text-classification/run_classification.py [hunk 1]
```diff
@@ -0,0 +1,770 @@
...
+
+    # Initialize our Trainer
+    trainer = ORTTrainer(
+        model=model,
+        args=training_args,
+        train_dataset=train_dataset if training_args.do_train else None,
+        eval_dataset=eval_dataset if training_args.do_eval else None,
+        compute_metrics=compute_metrics,
+        tokenizer=tokenizer,
+        data_collator=data_collator,
```
Comment (JingyaHuang) [inline examples/onnxruntime/training/text-classification/run_classification.py:698]: For `ORTTrainer` , the `feature` argument was mandatory. I opened a PR #1335 to make it optional.

File: examples/onnxruntime/training/text-classification/zero_stage_2.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (JingyaHuang) [inline examples/onnxruntime/training/text-classification/zero_stage_2.json:1]: Could you also elaborate in the readme or in the script (eg. with `deepspeed` argument of `ORTTrainer`) how this config file would be used?

**Review by JingyaHuang — COMMENTED** [review]
Hi @prathikr, thanks for adding the example, I put some minor nits. The classification example looks good to me, and I wonder is a causal language modeling fine-tune example be more interesting to the community if you want to promote ORT training with llama2.

**Comment by prathikr** [comment]
> Hi @prathikr, thanks for adding the example, I put some minor nits. The classification example looks good to me, and I wonder if a causal language modeling fine-tuning example could also be beneficial to the community since llama2 is a generative model.

@JingyaHuang run_clm.py is already available [here](https://github.com/huggingface/optimum/blob/main/examples/onnxruntime/training/language-modeling/run_clm.py)

**Comment by JingyaHuang** [comment]
Hi @prathikr, it's true that the training examples of ort training last updated a while ago. I will update them this week, but in long term, it would be hard for the team to keep examples always updated with transformers releases, the best way to automate the process would be to build two CIs in optimum repo:
* 1st CI: clone transformers repo and replace training examples with ORT APIs(`ORTTrainer` and `ORTTrainingArguments`) and open a pull request in the optimum repo.
* 2nd CI: to valid all updated examples (probably wth [`test_examples.py`](https://github.com/huggingface/optimum/blob/main/examples/onnxruntime/training/test_examples.py) which also need to be updated by adding all examples we have now) 

Ping @fxmarty for some inputs, probably you have a better idea.

**Comment by prathikr** [comment]
> Hi @prathikr, it's true that the training examples of ort training last updated a while ago. I will update them this week, but in long term, it would be hard for the team to keep examples always updated with transformers releases, the best way to automate the process would be to build two CIs in optimum repo:
> 
> * 1st CI: clone transformers repo and replace training examples with ORT APIs(`ORTTrainer` and `ORTTrainingArguments`) and open a pull request in the optimum repo.
> * 2nd CI: to valid all updated examples (probably wth [`test_examples.py`](https://github.com/huggingface/optimum/blob/main/examples/onnxruntime/training/test_examples.py) which also need to be updated by adding all examples we have now)
> 
> Ping @fxmarty for some inputs, probably you have a better idea.

Thanks @JingyaHuang for getting the ball rolling on this. We have internal pipelines that depends on optimum and transformers examples being the same, so this will be very useful. Can we follow up on this task in a separate issue/PR and merge this example for now?

**Review by fxmarty — APPROVED** [review]
LGTM!

**Pull request merged**